scripts/android/autopower/autopower.py

```diff
@@ -48,136 +48,152 @@ def cpuPath(cpuIndex):
-    current = collectAverageCurrent()
-    print "Result of measurement: Current = ", current
-
-#temprorarily, we just wait 10 second.
-#########
-#[TODO]need a mechanism to check synthmark has done.
-    time.sleep(10)
-
-# gather results from SynthMark, check utilization status
-# If Utilization is over 90%, we will skip lower frequecy case.
-    synthmarkOutput = subprocess.check_output(["adb", "shell", "cat", "/sdcard/result.txt"])
+    # launch SynthMark by test_once.sh, that is include actions as below
+    # 1. hold wakelock
+    # 2. set cpu_freq
+    # 3. simulate power key
+    # 4. exec synthmark
+    #######Input arg###########
+    # $1 = testMode, $2 = numVoices, $3 = delaySecond
+    # $4 = durationSecond, $5 = cpuIndex, $6 = frequency
+    # ex: adb shell "nohup /system/bin/test_once.sh u 100 5 15 7 1574400 &" &
+
+    # Make sure the old results are cleaned
+    adbCommand(["rm", "-f", "/sdcard/result.txt"])
+    command = ["nohup", "/system/bin/test_once.sh", testMode, str(numVoices), str(delaySecond)]
+    command = command + [str(durationSecond), str(cpuIndex), str(frequency), str(burstSize), "&> /dev/null &"]
+    # print "Running command: ", command
+
+    adbCommand(command)
+    current, var = collectAverageCurrent()
+
+    synthmarkOutput = ""
+    # Try for 20 seconds to read the results from synthmark
+    for i in range(10):
+        try:
+            synthmarkOutput = adbCommand(["cat", "/sdcard/result.txt"])
+            break
+        except subprocess.CalledProcessError:
+            # print "Sleeping till synthmark finishes"
+            time.sleep(2)
+
+    # If we never get it, we will assume we overloaded
+    # Gather results from SynthMark, check utilization status
+    # If Utilization is over 90%, we will skip lower frequecy case.
+    util = 100
     listSynthmark = synthmarkOutput.splitlines()
     for line in listSynthmark:
         if ("UtilizationMark = " ) in line:
-            print line
             util = float(line.strip('UtilizationMark = '))*100
-            if util > 90:
-                overload = True
-                break
-            else:
-                overload = False
-        else:
-            overload = False
+            break
 
-    return current, overload;
+    return current, var, util
 
-def measurePowerSeries(cpuIndex, numVoices):
-    print "Measure Power series:"
-    print "cpuIndex = ", cpuIndex
-    print "numVoices = ", numVoices
 
-# turn on USB passthrough
-    setupUSBCommand = ["python", monsoonPath, "--usbpassthrough", "auto"]
-    subprocess.call(setupUSBCommand)
+# This function turns passthrough on, runs the provided adb shell command, and
+# turns passthrough off.
+# @param String arr of commands to run through adb shell
+# @return Both the stdout and stderr of the commands run
+def adbCommand(stringArgs):
+    command = ["adb", "shell"] + stringArgs
+    # We mute stderr of the monsoon bc it is very noisy
```

Comment:
"bc"! is that like AC or DC? or "because"?

Reply:
Sorry, used to bc as an abbreviation of because, will clarify.